### PR TITLE
Rewrite without dolittle

### PR DESCRIPTION
**Pull request by srudaa**
## Summary

Rewrite the connector based new framework (https://github.com/RaaLabs/Edge)

### Changed

- Total rewrite to new framework

### CODE REVIEW

File: Source/Program.cs [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
+    [ExcludeFromCodeCoverage]
+    static class Program
     {
-        static void Main()
+        static void Main(string[] args)
+        {
+            Run().Wait();
+        }
+
+        private static async Task Run()
```
Comment (rafaelschlatter) [inline Source/Program.cs:20]: Is this preferred over doing  application setup and `application.Run().Wait();` directly in `Main`?

File: Source/Dockerfile [hunk 2]
```diff
@@ -1,42 +1,24 @@
-FROM microsoft/dotnet:2.1-runtime-stretch-slim as base
-
-ARG CONFIGURATION=Release
-
-RUN echo Configuration = $CONFIGURATION
-
-RUN if [ "$CONFIGURATION" = "Debug" ] ; then apt-get update && \
```
Comment (rafaelschlatter) [inline Source/Dockerfile:7]: Why was this if statement here previously? I saw the same in health monitor, but did not delete it.

File: Source/Events/KymaDatapointOutput.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    [OutputName("output")]
+    public class KymaDatapointOutput : IEdgeHubOutgoingEvent
+    {
+        /// <summary>
+        /// Represents the Source system.
+        /// </summary>
+        public string source { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tag. Represens the sensor name from the source system.
```
Comment (rafaelschlatter) [inline Source/Events/KymaDatapointOutput.cs:20]: Typo in docstring

File: Source/Events/KymaSentenceReceived.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        /// <param name="sentence"></param>
+        public KymaSentenceReceived(string sentence)
+        {
+            Sentence = sentence;
+        }
+        /// <summary>
+        /// Kyma sentence
+        /// </summary>
+        /// <param name="sentence"></param>
+        public static implicit operator KymaSentenceReceived(string sentence) => new KymaSentenceReceived(sentence);
```
Comment (rafaelschlatter) [inline Source/Events/KymaSentenceReceived.cs:30]: When is this operator used? Whats the difference between this and the constructor?

**Review by rafaelschlatter — APPROVED** [review]
Only minor comments, and some questions. I didnt look at the tests.

File: Source/Dockerfile (hunk 2 above)
Comment (srudaa) [reply Source/Dockerfile:7]: It is for debugging when testing locally with iotedgehubdev. Could be good idea to not delete this.

File: .dockerignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+**/obj
+**/bin
+Dockerfile
```
[the remark this answers is not in the document]
Comment (srudaa) [reply .dockerignore:3]: Agree

File: Source/Events/KymaSentenceReceived.cs (hunk 4 above)
Comment (srudaa) [reply Source/Events/KymaSentenceReceived.cs:30]: Got this from @thotokraa let us have a separate talk regarding this

**Pull request merged**